### PR TITLE
chore: infer SVG size from style

### PR DESCRIPTION
**Pull request by AmitShwarts**
## Description

allow SVG to infer size from the style object

## Changelog

Now SVG can infer size from style

### CODE REVIEW

File: src/components/svgImage/index.tsx [hunk 1]
```diff
@@ -28,7 +30,12 @@ function SvgImage(props: SvgImageProps) {
   if (isSvgUri(data)) {
     return <SvgCssUri {...others} uri={data.uri}/>;
   } else if (typeof data === 'string') {
-    return <SvgXml xml={data} {...others}/>;
+    const flattenStyle = StyleSheet.flatten(props.style) as Record<string, any>;
```
Comment (M-i-k-e-l) [inline src/components/svgImage/index.tsx:33]: We use `flatten` very rarely, why is this required here?

Comment (AmitShwarts) [reply src/components/svgImage/index.tsx:33]: to extract the `width` and `height` from the provided styles.
do you have suggestion how to do it differently?

Comment (OrenZak) [reply src/components/svgImage/index.tsx:33]: are we sure its coming with more than one style object ?
if not, and we always have just single style object (and not as array).?
we can remove that.

Comment (AmitShwarts) [reply src/components/svgImage/index.tsx:33]: yes i'm sure. we can't remove it

Comment (M-i-k-e-l) [reply src/components/svgImage/index.tsx:33]: Can you please provide the scenario this is used in?

Comment (AmitShwarts) [reply src/components/svgImage/index.tsx:33]: Yes, for example when using Button with SVG Icon, we're passing `iconStyle` with `width` & `height`.
in this case it's been passed to the SVG component, and without this PR the height and width is not respected.

Comment (AmitShwarts) [reply src/components/svgImage/index.tsx:33]:
```
    <Button
      id={id}
      testID={TestIds.buttonElement}
      style={style}
      color={color}
      iconSource={iconSource}
      iconStyle={{width: Spacings.s6, height: Spacings.s6}}
      round
      size={Button.sizes.large}
    />
```

Comment (M-i-k-e-l) [reply src/components/svgImage/index.tsx:33]: As discussed, lets allow passing `iconProps` to the `Button` instead, which will allow sending `width` and `height`

**Review by M-i-k-e-l — APPROVED** [review]
